WebService/v2/Models/DeviceModelApiModel/DeviceModelSimulationScript.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v2.Exceptions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v2.Models.DeviceModelApiModel
+{
+    public class DeviceModelSimulationScript
+    {
+        [JsonProperty(PropertyName = "Type")]
+        public string Type { get; set; }
+
+        [JsonProperty(PropertyName = "Path")]
+        public string Path { get; set; }
+
+        [JsonProperty(PropertyName = "Params", NullValueHandling = NullValueHandling.Ignore)]
+        public object Params { get; set; }
+
+        public DeviceModelSimulationScript()
+        {
+            this.Type = string.Empty;
+            this.Path = string.Empty;
+            this.Params = null;
+        }
+
+        // Map API model to service model
+        public Script ToServiceModel()
+        {
+            if (this.IsEmpty()) return null;
+
+            return new Script
+            {
+                Type = !string.IsNullOrEmpty(this.Type) ? this.Type : null,
+                Path = !string.IsNullOrEmpty(this.Path) ? this.Path : null,
+                Params = this.Params
+            };
+        }
+
+        // Map service model to API model
+        public static DeviceModelSimulationScript FromServiceModel(Script value)
+        {
+            if (value == null) return null;
+
+            return new DeviceModelSimulationScript
+            {
+                Type = string.IsNullOrEmpty(value.Type) ? null : value.Type,
+                Path = string.IsNullOrEmpty(value.Path) ? null : value.Path,
+                Params = value.Params
+            };
+        }
+
+        public bool IsEmpty()
+        {
+            return string.IsNullOrEmpty(this.Type)
+                   && string.IsNullOrEmpty(this.Path)
+                   && this.Params == null;
+        }
+
+        public void ValidateInputRequest(ILogger log)
+        {
+            const string INTERNAL = "internal";
+            const string NO_TYPE = "Script must contains a valid type";
+            const string NO_PATH = "Script must contains a valid path";
+
+            if (string.IsNullOrEmpty(this.Type))
+            {
+                log.Error(NO_TYPE, () => new { Script = this });
+                throw new BadRequestException(NO_TYPE);
+            }
+
+            if (string.IsNullOrEmpty(this.Path))
+            {
+                log.Error(NO_PATH, () => new { Script = this });
+                throw new BadRequestException(NO_PATH);
+            }
+
+            if (this.Type == INTERNAL)
+            {
+                this.ValidateParams(log);
+            }
+        }
+
+        private void ValidateParams(ILogger log)
```

Comment:
rename to 'ValidateInternalScriptParams'

Reply:
also add a comment showing an example of this block, something like
```
{
    "Type": "internal",
    "Path": "math.random.withinrange",
    "Params": {
        "temperature": {
            "Min"  : "1",
            "Max"  : "12",
            "Step" : 1,
            "Unit" : "C"
        },
        "pressure": {
            "Min"  : "10",
            "Max"  : "90",
            "Step" : 1,
            "Unit" : "psi"
        },
    }
}
```

Reply:
Done.